### PR TITLE
Add read-only mode in the SDC library

### LINKED ISSUE #1704: Read-only summary/review page
**Is your feature request related to a problem? Please describe.**
The review page currently has an "EDIT" button that allows the user to go back to the questionnaire. But there is a need for just having a read-only review/summary page in certain use cases. 

**Describe the solution you'd like**
Create a new setting to allow for the review/summary mode to be readonly

**Describe alternatives you've considered**
We could also render the normal questionnaire and make all questions readonly. But that seems to be unnecessary. Why do we render the widgets if they can never be editted anyway?

**Additional context**
NA

**Would you like to work on the issue?**
NA

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1704 

**Description**
Add read-only mode in the SDC library.

- introduce a new argument to the questionnaire fragment `QuestionnaireFragment.EXTRA_READ_ONLY`
- setting this argument to true while launching the questionnaire fragent will set the questionnaire in read-only mode
- read-only mode will look like review mode but without the option to edit the answers

**Alternative(s) considered**
NA

**Type**
Feature

**Screenshots (if applicable)**
NA

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt [hunk 1]
```diff
@@ -3265,12 +3308,16 @@ class QuestionnaireViewModelTest {
...
+          (viewModel.questionnaireStateFlow.first().displayMode as DisplayMode.EditMode)
+            .pagination.showReviewButton
+        )
+        .isTrue()
     }
   }
 
   @Test
-  fun `toggle review mode to true should not show review button`() {
+  fun `toggle review mode to true should not show edit button`() {
```
Comment (omarismail94) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt:3320]: aren't we asserting that the edit button is visible in this test? if so, let's remove change the test name from 
`toggle review mode to true should not show edit button` to 
`toggle review mode to true should show edit button only`

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt [hunk 2]
```diff
@@ -357,23 +353,23 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
...
   internal fun setShowSubmitButtonFlag(showSubmitButton: Boolean) {
-    showSubmitButtonFlag = showSubmitButton
+    this.shouldShowSubmitButton = showSubmitButton
   }
 
   /** [QuestionnaireState] to be displayed in the UI. */
   internal val questionnaireStateFlow: StateFlow<QuestionnaireState> =
-    combine(modificationCount, currentPageIndexFlow, reviewFlow) { _, _, reviewMode ->
-        getQuestionnaireState(reviewMode = reviewMode)
+    combine(modificationCount, currentPageIndexFlow, isInReviewModeFlow) { _, _, reviewMode ->
```
Comment (omarismail94) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt:365]: as `getQuestionnaireState` no longer needs input parameters, do we need to have `reviewMode` ? Let's change it to `_`

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt [hunk 3]
```diff
@@ -810,12 +809,15 @@ typealias ItemToParentMap =
 internal data class QuestionnaireState(
   /** The items that should be currently-rendered into the Fragment. */
   val items: List<QuestionnaireItemViewItem>,
-  /** The pagination state of the questionnaire. */
-  val pagination: QuestionnairePagination,
   /** Tracks reviewMode in order to update the UI. */
```
Comment (omarismail94) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt:812]: nit: reviewMode -> displayMode

**Review by omarismail94 — COMMENTED** [review]
Nice, clean change Jing! Left a few minor comments

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt [hunk 4]
```diff
@@ -247,6 +269,12 @@ open class QuestionnaireFragment : Fragment() {
      */
     const val EXTRA_SHOW_REVIEW_PAGE_FIRST = "show-review-page-first"
 
+    /**
+     * An [Boolean] extra to control if the questionnaire is read-only. If review page and read-only
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt:273]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt (hunk 1 above)
Comment (jingtang10) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt:3320]: done.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt (hunk 2 above)
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt:365]: done.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt (hunk 3 above)
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt:812]: the comment adds no value i feel so i removed it....

**Review by omarismail94 — APPROVED** [review]

**Comment by jingtang10** [comment]
thx omar!

**Pull request merged**